### PR TITLE
Fixes the properties for Mixpanel Actions

### PR DESCRIPTION
**Pull request by zihejia**
This PR fixes the following issues for Mixpanel Actions:
- [fixes for the timestamp of Mixpanel event](https://github.com/segmentio/action-destinations/commit/0a1e38104deb40ebf9e38d254e05ddbde4e91211) , make it optional and will prevent it (along with some other mixpanel properties) being overridden 
- [automatically derive 'name'  from 'firstName' and 'lastName'](https://github.com/segmentio/action-destinations/commit/5d13d62b7e329d47b079a601bb6ce34d6678b25f)

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/mixpanel/trackEvent/index.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
-import { ActionDefinition, RequestClient } from '@segment/actions-core'
+import { ActionDefinition, RequestClient, omit } from '@segment/actions-core'
 import type { Settings } from '../generated-types'
 import type { Payload } from './generated-types'
 import { MixpanelEvent } from '../mixpanel-types'
 import { getApiServerUrl, getBrowser, getBrowserVersion, cheapGuid } from '../utils'
 import dayjs from '../../../lib/dayjs'
+import { eventProperties } from '../mixpanel-properties'
+
+const mixpanelReservedProperties = ['time', 'id', '$anon_id', 'distinct_id', '$group_id', '$insert_id', '$user_id']
```
Comment (jaredmixpanel) [inline packages/destination-actions/src/destinations/mixpanel/trackEvent/index.ts:9]: `id` is a mixpanel reserved prop?

File: packages/destination-actions/src/destinations/mixpanel/trackEvent/index.ts [hunk 2]
```diff
@@ -111,324 +114,7 @@ const action: ActionDefinition<Settings, Payload> = {
...
-      default: {
-        '@path': '$.context.userAgent'
-      }
-    },
-    enable_batching: {
-      type: 'boolean',
-      label: 'Batch Data to Mixpanel',
-      description: 'Set as true to ensure Segment sends data to Mixpanel in batches.',
-      default: true
-    }
```
Comment (jaredmixpanel) [inline packages/destination-actions/src/destinations/mixpanel/trackEvent/index.ts:431]: where did all of this go?

Comment (zihejia) [reply packages/destination-actions/src/destinations/mixpanel/trackEvent/index.ts:431]: in `mixpanel-properties.ts`

File: packages/destination-actions/src/destinations/mixpanel/trackEvent/index.ts (hunk 1 above)
Comment (zihejia) [reply packages/destination-actions/src/destinations/mixpanel/trackEvent/index.ts:9]: `id` is provided by the classic integration, this is for backward compatibility and we don't want the user to overwrite this prop.

File: packages/destination-actions/src/destinations/mixpanel/mixpanel-types.ts [hunk 3]
```diff
@@ -15,7 +15,6 @@ export type MixpanelEventProperties = {
   $device_type?: string // 'ios'
   $device_name?: string // 'maguro'
   $group_id?: string // 'groupId123'
-  $identified_id?: string // 'user1234'
```
[the remark this answers is not in the document]
Comment (zihejia) [reply packages/destination-actions/src/destinations/mixpanel/mixpanel-types.ts:18]: it should not be in the Track calls.

[the remark this answers is not in the document]
Comment (zihejia) [reply packages/destination-actions/src/destinations/mixpanel/mixpanel-types.ts:18]: will undo the change, thanks for asking

**Review by brennan — APPROVED** [review]

**Pull request merged**